sdk/communication/azure-communication-callingserver/src/main/java/com/azure/communication/callingserver/CallAutomationAsyncClient.java

```diff
@@ -0,0 +1,305 @@
+     * @throws RuntimeException all other wrapped checked exceptions if the request fails to be sent.
+     * @return Response for a successful CreateCallConnection request.
+     */
+    @ServiceMethod(returns = ReturnType.SINGLE)
+    public Mono<Response<CallConnectionProperties>> createCallWithResponse(CreateCallOptions createCallOptions) {
+        return withContext(context -> createCallWithResponseInternal(createCallOptions, context));
+    }
+
+    Mono<Response<CallConnectionProperties>> createCallWithResponseInternal(CreateCallOptions createCallOptions,
+                                                                            Context context) {
+        try {
+            context = context == null ? Context.NONE : context;
+            List<CommunicationIdentifierModel> targetsModel = createCallOptions.getTargets()
+                .stream().map(CommunicationIdentifierConverter::convert).collect(Collectors.toList());
+
+            CallSourceInternal callSourceDto = new CallSourceInternal().setIdentifier(
+                CommunicationIdentifierConverter.convert(createCallOptions.getSource()));
+            if (createCallOptions.getSourceCallerId() != null) {
+                callSourceDto.setCallerId(new PhoneNumberIdentifierModel().setValue(createCallOptions.getSourceCallerId()));
+            }
+
+            CreateCallRequestInternal request = new CreateCallRequestInternal()
+                .setSource(callSourceDto)
+                .setTargets(targetsModel)
+                .setCallbackUri(createCallOptions.getCallbackUri().toString())
+                .setSubject(createCallOptions.getSubject());
+
+            return serverCallingInternal.createCallWithResponseAsync(request, context)
+                .onErrorMap(HttpResponseException.class, ErrorConstructorProxy::create)
+                .map(response -> {
+                    try {
+                        return new SimpleResponse<>(response, CallConnectionPropertiesConstructorProxy.create(response.getValue()));
+                    } catch (URISyntaxException e) {
+                        logger.logExceptionAsError(new RuntimeException(e));
+                        return null;
+                    }
+                });
+        } catch (RuntimeException ex) {
+            return monoError(logger, ex);
+        }
+    }
+
+    /**
+     * Answer an incoming call
+     *
+     * @param incomingCallContext The incoming call context.
+     * @param callbackUri The call back uri. Optional
+     * @throws CallingServerErrorException thrown if the request is rejected by server.
+     * @throws RuntimeException all other wrapped checked exceptions if the request fails to be sent.
+     * @return Response for a successful CreateCallConnection request.
+     */
+    @ServiceMethod(returns = ReturnType.SINGLE)
+    public Mono<CallConnectionProperties> answerCall(String incomingCallContext, String callbackUri) {
```

Comment:
AnswerCall from .NET returns CallProperties + CallConnection(client) in AnswerCallResult. Does this apply here?

Reply:
So far we haven't received any feedback(neither on the review nor on apiView comments) on this yet, I wonder if we need to change or not as well.
I am planning to merge this first, and then we can confirm in the next review.

Reply:
Shouldn't we keep SDKs similar as possible so developers don't have to relearn SDK on each language?

Reply:
Agree, fixed